pinot-core/src/main/java/org/apache/pinot/server/realtime/ControllerLeaderLocator.java

```diff
@@ -59,21 +61,23 @@ public class ControllerLeaderLocator {
 
   ControllerLeaderLocator(HelixManager helixManager) {
     _helixManager = helixManager;
-    _cachedControllerLeaderMap = new HashMap<>();
+    _cachedControllerLeaderMap = new ConcurrentHashMap<>();
   }
 
   /**
    * To be called once when the server starts
    * @param helixManager should already be started
    */
   public static void create(HelixManager helixManager) {
-    if (_instance != null) {
-      // We create multiple server instances in the hybrid cluster integration tests, so allow the call to create an
-      // instance even if there is already one.
-      LOGGER.warn("Already created");
-      return;
+    if (_instance == null) {
```

Comment:
This is called only once during server starts, which is always single threaded. I don't think we need to synchronize this